sklearn/utils/estimator_checks.py

```diff
@@ -2359,6 +2362,89 @@ def check_classifiers_regression_target(name, estimator_orig):
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_estimator_sparse_dense(name, estimator_orig):
+    rng = np.random.RandomState(52)
+    X = rng.rand(40, 10)
+    X[X < .8] = 0
+    X_csr = sparse.csr_matrix(X)
+    y = (4 * rng.rand(40)).astype(np.int)
+    estimator = clone(estimator_orig)
+    estimator_sp = clone(estimator_orig)
+    tags = _safe_tags(estimator_orig)
+    if "multioutput" in tags:
+        y = y[:, np.newaxis]
+
+    for sparse_format in ['csr', 'csc', 'dok', 'lil', 'coo', 'dia', 'bsr']:
+        X_sp = X_csr.asformat(sparse_format)
+        # catch deprecation warnings
+        with ignore_warnings(category=DeprecationWarning):
+            if name in ['Scaler', 'StandardScaler']:
+                estimator.set_params(with_mean=False)
+                estimator_sp.set_params(with_mean=False)
+            if name in ['RobustScaler']:
+                # XXX naming is not consistent (with_mean vs with_centering) :(
+                estimator.set_params(with_centering=False)
+                estimator_sp.set_params(with_centering=False)
+            if name in ['TransformedTargetRegressor']:
+                estimator.set_params(regressor=LinearRegression(
+                    fit_intercept=False))
+                estimator_sp.set_params(regressor=LinearRegression(
+                    fit_intercept=False))
+        dense_vs_sparse_additional_params = defaultdict(dict,
+                {'Ridge': {'solver': 'cholesky'},
+                 })
+        params = dense_vs_sparse_additional_params[
+            estimator.__class__.__name__]
+        estimator.set_params(**params)
+        estimator_sp.set_params(**params)
+        if isinstance(estimator, NeighborsBase):
+            estimator.set_params(algorithm='brute')
+            estimator_sp.set_params(algorithm='brute')
+        for estimator_attr in ['base_estimator', 'estimator']:
+            if (hasattr(estimator, estimator_attr) and
+                    getattr(estimator, estimator_attr) is not None and
+                    'fit_intercept' in
+                    getattr(estimator, estimator_attr).get_params()):
+                getattr(estimator, estimator_attr).set_params(
+                    fit_intercept=False)
+                getattr(estimator_sp, estimator_attr).set_params(
+                    fit_intercept=False)
+        set_random_state(estimator)
+        set_random_state(estimator_sp)
+        X = pairwise_estimator_convert_X(X, estimator)
+        X_sp = pairwise_estimator_convert_X(X_sp, estimator)
+        try:
+            with ignore_warnings(category=DeprecationWarning):
+                estimator_sp.fit(X_sp, y)
+                estimator.fit(X, y)
+            if hasattr(estimator, "predict"):
+                pred = estimator.predict(X)
+                pred_sp = estimator_sp.predict(X_sp)
+                assert_array_almost_equal(pred, pred_sp, 2)
+                assert_equal(pred.shape, pred_sp.shape)
```

Comment:
It's better to use `assert_all_close`
and `assert ... == ...`

Reply:
Thanks, will do